dbt-snowflake/src/dbt/include/snowflake/macros/materializations/functions/scalar.sql

```diff
@@ -10,3 +11,14 @@
        {{ model.compiled_code }}
     $$
 {% endmacro %}
+
+{% macro snowflake__scalar_function_volatility_sql() %}
+    {% if model.config.get('volatility') == 'deterministic' %}
+        IMMUTABLE
+    {% elif model.config.get('volatility') == 'stable' %}
+        {% do exceptions.warn("`Stable` function volatility is not supported by Snowflake, and will be ignored") %}
+    {% elif model.config.get('volatility') == 'non-deterministic' %}
+        VOLATILE
+    {% endif %}
```

Comment:
is this more future proof?
```suggestion
    {% set volatility = model.config.get('volatility') %}
    {% if volatility == 'deterministic' %}
        IMMUTABLE
    {% elif model.config.get('volatility') == 'non-deterministic' %}
        VOLATILE
     {% elif volatility is not none %}
        {% do exceptions.warn("user passed unsupported volatility config value: " ~ volatility ~"; it will be ignored") %}
    {% endif %}
```

Reply:
Okay I think I've now added logic to sufficiently handle this 🙂